language/slackbot/demo_bot.js

```diff
@@ -0,0 +1,208 @@
+/* *****************************************************************************
+Copyright 2016 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+********************************************************************************
+
+This is a Slack bot built using the Botkit library (http://howdy.ai/botkit). It
+runs on a Kubernetes cluster, and uses one of the Google Cloud Platform's ML
+APIs, the Natural Language API, to interact in a Slack channel. It does this in
+two respects.
+
+First, it uses the NL API to assess the "sentiment" of any message posted to
+the channel, and if the positive or negative magnitude of the statement is
+sufficiently large, it sends a 'thumbs up' or 'thumbs down' in reaction.
+
+Second, it uses the NL API to identify the 'entities' in each posted message,
+and tracks them in a database (using sqlite3).  Then, at any time you can
+query the NL slackbot to ask it for the top N entities used in the channel.
+
+The README walks through how to run the NL slackbot as an app on a
+Google Container Engine/Kubernetes cluster, but you can also just run the bot
+locally if you like.
+To do this, create a file containing your Slack token (as described in
+the README), then point 'slack_token_path' to that file when you run the script:
+
+    echo my-slack-token > slack-token
+    slack_token_path=./slack-token node demo_bot.js
+
+See the README.md in this directory for more information about setup and usage.
+
+*/
+
+'use strict';
+
+const Language = require('@google-cloud/language');
+const Storage = require('@google-cloud/storage');
+
+
+var Botkit = require('botkit');
+var fs = require('fs');
+
+var controller = Botkit.slackbot({debug: false});
+
+var sqlite3 = require('sqlite3').verbose();
+// create our database if it does not already exist.
+var db = new sqlite3.Database('slackDB.db');
+
+const util = require('util');
+
+// the number of most frequent entities to retrieve from the db on request.
+const NumEnts = 20;
+// The magnitude of sentiment of a posted text above which the bot will respond.
+const SentimentThresh = 30;
+
+if (!process.env.slack_token_path) {
```

Comment:
NIT: Environment variables are typically all caps (at least that's the convention used throughout `nodejs-docs-samples`)